File: app/src/main/java/com/buzzybrains/mvvmarchitecture/model/Note.java
package com.buzzybrains.mvvmarchitecture.model;


import java.io.Serializable;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "note_table")
public class Note implements Serializable {

    @PrimaryKey(autoGenerate = true)
    private long id;

    private String title;

    private String description;

    private int priority;

    private boolean isSync;

    public Note() {
    }

    public Note(String title, String description, int priority, boolean isSync) {
        this.title = title;
        this.description = description;
        this.priority = priority;
        this.isSync = isSync;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getPriority() {
        return priority;
    }

    public void setPriority(int priority) {
        this.priority = priority;
    }


    public boolean isSync() {
        return isSync;
    }

    public void setSync(boolean sync) {
        isSync = sync;
    }


    @Override
    public String toString() {
        return "Note{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", priority=" + priority +
                ", isSync=" + isSync +
                '}';
    }
}

File: app/src/main/java/com/buzzybrains/mvvmarchitecture/model/Note.kt
package com.buzzybrains.mvvmarchitecture.model


import java.io.Serializable

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "note_table")
 data class Note (var title: String? = null,
             var description: String? = null,
             var priority: Int = 0 ,
             var isSync: Boolean = false): Serializable {

    @PrimaryKey(autoGenerate = true) var id: Long = 0


    override fun toString(): String {
        return "Note{" +
                "id=" + id +
                ", title='" + title + '\''.toString() +
                ", description='" + description + '\''.toString() +
                ", priority=" + priority +
                ", isSync=" + isSync +
                '}'.toString()
    }
}
